src/app/core/components/Navbar/Navbar.tsx

```diff
@@ -31,68 +28,321 @@ interface NavbarProps {
+      setTypingTimerID(id);
+    } else {
+      setSearchResult([]);
+    }
   };
 
-  const onInviteMemberClick = (): void => {
-    props.dispatch(uiActions.setIsInviteMemberDialogOpen(true));
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (searchResult.length > 0) {
+      setPreventBlur(false);
+      openItem(searchResult[selectedResult].id);
+    } else {
+      setLoadingSearch(true);
+      search();
+    }
   };
 
-  const onSearchInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
-    if (validationService.validateSearchText(e.target.value)) {
-      props.dispatch(
-        storageActions.setFilters({
-          text: e.target.value,
-        }),
-      );
+  const handleKeyDown = (e) => {
+    let item: number | null = null;
+    if (e.key === 'Escape') {
+      setPreventBlur(false);
+    } else if (e.key === 'ArrowUp') {
+      e.preventDefault();
+      setSelectedResult((current) => (current > 0 ? current - 1 : 0));
+      item = selectedResult > 0 ? selectedResult - 1 : 0;
+    } else if (e.key === 'ArrowDown') {
+      e.preventDefault();
+      setSelectedResult((current) => (current < searchResult.length - 1 ? current + 1 : searchResult.length - 1));
+      item = selectedResult < searchResult.length - 1 ? selectedResult + 1 : searchResult.length - 1;
     }
+    if (item) document.querySelector(`#searchResult_${item}`)?.scrollIntoView();
   };
 
-  const { user, storageFilters, hideSearch } = props;
-  if (!user) throw new Error('User is not defined');
+  const FilterItem = ({ id, Icon, name }) => (
+    <div
+      className={`${
+        filters.includes(id)
+          ? 'bg-primary bg-opacity-10 text-primary ring-primary ring-opacity-20'
+          : 'bg-white text-gray-80 ring-gray-10 hover:bg-gray-1 hover:shadow-sm hover:ring-gray-20 active:bg-gray-5 active:ring-gray-30'
+      } flex h-8 cursor-pointer items-center space-x-2 rounded-full px-3 font-medium shadow-sm ring-1 transition-all duration-100 ease-out`}
+      onClick={() => toggleFilter(id)}
+    >
+      <Icon className="h-5 w-5 drop-shadow-sm filter" />
+      <span className="text-sm">{name}</span>
+    </div>
+  );
+
+  const filterItems = [
+    {
+      id: 'folder',
+      Icon: iconService.getItemIcon(true),
+      name: translate('general.searchBar.filters.folder') as string,
+    },
+    {
+      id: 'pdf',
+      Icon: iconService.getItemIcon(false, 'pdf'),
+      name: translate('general.searchBar.filters.pdf') as string,
+    },
+    {
+      id: 'image',
+      Icon: iconService.getItemIcon(false, 'jpg'),
+      name: translate('general.searchBar.filters.image') as string,
+    },
+    {
+      id: 'video',
+      Icon: iconService.getItemIcon(false, 'mp4'),
+      name: translate('general.searchBar.filters.video') as string,
+    },
+    {
+      id: 'audio',
+      Icon: iconService.getItemIcon(false, 'mp3'),
+      name: translate('general.searchBar.filters.audio') as string,
```

Comment:
Why does each function `translate` the casting to string? I don't think it is necessary since the `translate` function returns a string